pkgs/by-name/dc/dc3dd/package.nix

```diff
@@ -0,0 +1,48 @@
+{ lib
+, stdenv
+, fetchzip
+, perl538Packages
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "dc3dd";
+  version = "7.3.1";
+
+  src = fetchzip {
+      url = "http://downloads.sf.net/sourceforge/dc3dd/dc3dd-${finalAttrs.version}.zip";
+      hash = "sha256-SYDoqGlsROHX1a0jJX11F+yp6CeFK+tZbYOOnScC6Ig=";
+  };
+
+  postPatch = ''
```

Comment:
```suggestion
  preConfigure = ''
```

Reply:
I see some packages use `postPatch` and [other ones](https://github.com/NixOS/nixpkgs/blob/6189bca1cc94e7a8be39eb611b942910355d9555/pkgs/development/compilers/openjdk/17.nix#L77) use `preConfigure`. So, I guess `postPatch` should be practically used for changing code content of source files and `preConfigure` to execute bash commands on source files?

Reply:
In Nixpkgs, the build process is split in phases.
The patch phase is responsible for the task of patching the source files.
The configure phase is responsible for the task of configuring the package.